### ISSUE TITLE
SteamCMD doesn't work

### ISSUE THREAD

**Issue by Wolly300** [jkread/arkserver#5]
I used nearly the same docker-compose as given but the log gives me the following error:

![image](https://user-images.githubusercontent.com/43980888/107892459-79a10180-6f25-11eb-958b-d879dd3e66a8.png)

**Comment by motey** [comment]
Same here, looks like SteamCMD installation is broken in the current "latest" image.

**Comment by motey** [comment]
`docker run -it ark /bin/bash` # with local self build  jkread/arkserver image 
`$ steamcmd`

actually works. must be an issue with how steamcmd is installed and how https://github.com/arkmanager/ark-server-tools expects it to be setup

**Comment by motey** [comment]
seems like steamcmd needs a initial run to update/install. i dont know if thats a new behavior and therefore breaking the current image.
When running a `steamcmd +quit` before calling arkmanager the error seems to be solved and the server starts up.
https://github.com/motey/arkserver/commit/b8ae5dadb83bff5430426e794c0afbed1e4cd1ed

**Comment by motey** [comment]
pull request https://github.com/jkread/arkserver/pull/6

**Comment by Wolly300** [comment]
> pull request #6

I tried to run your bugfix. Built the new Dockerfile with the `steamcmd +quit` line, but got an permission denied error in my docker compose.

![image](https://user-images.githubusercontent.com/43980888/108002344-a3265f80-6fef-11eb-91da-ba93faa248d5.png)

@motey nice job, but please help me I'm not a god in git/docker 😂😂

**Comment by motey** [comment]
mh, weird, worked for me. time for bed :sleepy: 

i'll test it again tomorrow and create a temp image (until we have a real fix) and push it to dockerhub for you.

**Comment by Wolly300** [comment]
> mh, weird, worked for me. time for bed 😪
> 
> i'll test it again tomorrow and create a temp image (until we have a real fix) and push it to dockerhub for you.

Yes, it's one in the morning, thank you for the help.

In my case it seems to be a issue with uid and gid.

**Comment by motey** [comment]
@Wolly300  There are some more issues with the steamcmd in this image. i cant get the server to be listening, which may or not may be related to steamcmd issue. atm i am looking for alternative ark server docker images :( 
https://hub.docker.com/r/colakatz/ark-docker looks promising

**Comment by Wolly300** [comment]
> @Wolly300 There are some more issues with the steamcmd in this image. i cant get the server to be listening, which may or not may be related to steamcmd issue. atm i am looking for alternative ark server docker images :(
> https://hub.docker.com/r/colakatz/ark-docker looks promising

@motey Is your server running with the new image and can you join in ARK?

**Comment by Wolly300** [comment]
> @Wolly300 There are some more issues with the steamcmd in this image. i cant get the server to be listening, which may or not may be related to steamcmd issue. atm i am looking for alternative ark server docker images :(
> https://hub.docker.com/r/colakatz/ark-docker looks promising

@motey This Image runs successfully but I couldn‘t join in ARK.

https://github.com/TuRz4m/Ark-docker

**Comment by motey** [comment]
No,. as i said 

>  i cant get the server to be listening,

i will report to you here if i found a solution

**Comment by motey** [comment]
@Wolly300 https://github.com/motey/Ark-docker Created a fork with docker image, which is working for me :) give it a try

**Comment by Wolly300** [comment]
> @Wolly300 https://github.com/motey/Ark-docker Created a fork with docker image, which is working for me :) give it a try

The server is running, but I can't join

###########################################################################
# Ark Server -  Tue Feb 16 19:57:13 CET 2021
# UID 1000 - GID 1000
###########################################################################
No game files found. Installing...
Running command 'install' for instance 'main'
Installing ARK server ...  -/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/ No crontab set.
Running command 'start' for instance 'main'
Checking for updates before starting
Checking for update; PID: 2344
Your server is already up to date! The most recent version is 6228959.
The server is starting...
The server is now running, and should be up within 10 minutes
Waiting...
###########################################################################

**Comment by motey** [comment]
looks like you have another issue :)

lets have a chat @ https://bloochat.com/2gj6f maybe i can help you

**Comment by jkread** [comment]
I've not had an issue but I'll test this with a new c;lean build and add the init first if needed.

**Comment by jkread** [comment]
I've added an init/quit in the dockerbuild. However I need someone to test, mine were successful but I couldn't recreate the issue beforehand.

**Comment by Wolly300** [comment]
I solved this problem, it was a combined (docker / ufw) problem with the port forwarding for port 7777 and 7778.

But this Log output is still ugly:

Installing ARK server ...  -/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\-/|\- No crontab set.

**Comment by jkread** [comment]
Yeah the install logging all comes out of arkmanager or something it calls.

**Comment by jkread** [comment]
I believe this to be resolved in the current latest build. Please test and let me know. May require a clearing of volumes or binds.

**Comment by Wolly300** [comment]
I didn't test it. But you should use this image as sample. 
I recently reported a crontab issue.

https://github.com/motey/Ark-docker

https://github.com/motey/Ark-docker/issues/1

**Comment by jkread** [comment]
Yeah, I tried to use https://github.com/TuRz4m/Ark-docker that is based on before but it had more issues than I cared to deal with. Any way I've fixed all the issues I've encountered or been made aware of with this version. I'll be using it for my instances and will support any issues that anyone finds.
